leetcode/621_task_scheduler.py

```diff
@@ -0,0 +1,63 @@
+from typing import List
+from collections import Counter
+from collections import deque
+import heapq
+
+class Solution:
+    def leastInterval(self, tasks: List[str], n: int) -> int:
+        """
+        Calculates the minimum time required to complete all tasks with a cooldown of `n`
+        units of time between two identical tasks.
+
+        Args:
+            tasks (List[str]): A list of tasks represented by uppercase letters ('A' to 'Z').
+            n (int): The cooldown period between two identical tasks.
+
+        Returns:
+            int: The minimum time units required to complete all tasks.
+        """
+
+        freq = [0] * 26
+        for task in tasks:
+            freq[ord(task) - ord('A')] += 1
+
+        max_freq = max(freq)
+        freq_count = freq.count(max_freq)
+
+        total_gaps = n * (max_freq - 1)
+        fillers_available = len(tasks) - max_freq
+
+        idles = total_gaps - fillers_available + (freq_count - 1)
+
+        return len(tasks) + max(0,idles)
+
+    def leastInterval_heap(self, tasks: List[str], n: int) -> int:
+        """
+        Calculates the minimum time required to complete all tasks with a cooldown of `n`
+        units of time between two identical tasks using a max-heap approach.
+
+        Args:
+            tasks (List[str]): A list of tasks represented by uppercase letters ('A' to 'Z').
+            n (int): The cooldown period between two identical tasks.
+
+        Returns:
+            int: The minimum time units required to complete all tasks.
+        """
+
+        count = Counter(tasks)
+        maxHeap = [-cnt for cnt in count.values()]
```

Comment:
It's important to stay consistent with styling. Stick to `snake_case` instead of `camelCase`